app/src/main/java/org/fossasia/susi/ai/activities/MainActivity.java

```diff
@@ -741,46 +734,58 @@ public void execute(Realm bgRealm) {
                                         rvChatFeed.getRecycledViewPool().clear();
                                         recyclerAdapter.notifyItemChanged((int) id);
                                         final String setMessage = answer;
-
+                                        final int counterValue = counter;
+                                        actionType = response.body().getAnswers().get(0).getActions().get(counter).getType();
                                         final Handler delayHandler = new Handler();
                                         delayHandler.postDelayed(new Runnable() {
                                             @Override
                                             public void run() {
-                                                addNewMessage(setMessage, ismap, isHavingLink, isPieChart, isSearchResult, datumList);
+                                                actionType = susiResponse.getAnswers().get(0).getActions().get(counterValue).getType();
```

Comment:
@samsruti Actually, issue #372 was to find location of user and add it as a parameters in susi api call. Issue #419 arrived due to wrong interpretation of susi response. So, there isn't anything in this PR which resolves anything from issue #372 . So, you may remove it from title of this PR.